### PR TITLE
fix(selector): normalise whitespaces in text selector

### LINKED ISSUE #3830: [Feature] Text selector engine normalise whitespaces
In the web people often use `&nbsp;`. If they they want to match the example (see below) with a normal whitespace, it would not work since its a non-breaking whitespace.

```js
// @ts-check
const playwright = require("playwright");

(async () => {
  const browser = await playwright.chromium.launch();
  const page = await browser.newPage();
  await page.setContent(`<button>Sign&nbsp;in</button>`)
  console.log(await page.innerText("text=Sign in"))
  await browser.close();
})();
```

Proposal: Normalise whitespaces before we compare them.

### PR DESCRIPTION
**Pull request by mxschmitt**
Fixes #3830

### CODE REVIEW

File: src/server/injected/textSelectorEngine.ts [hunk 1]
```diff
@@ -118,7 +118,7 @@ function queryInternal(root: SelectorRoot, matcher: Matcher, shadow: boolean): E
     if (!node)
       break;
     if (node.nodeType === Node.TEXT_NODE) {
-      lastText += node.nodeValue;
+      lastText += node.nodeValue ? node.nodeValue.replace(/\s/g, ' ') : node.nodeValue;
```
Comment (dgozman) [inline src/server/injected/textSelectorEngine.ts:121]: Let's instead update `matcher` to cover `element.value` check below, and also similar checks in queryAllInternal.

File: test/selectors-text.spec.ts [hunk 2]
```diff
@@ -94,6 +94,9 @@ it('query', async ({page, isWebKit}) => {
   });
   expect(await page.$eval(`text=lowo`, e => e.outerHTML)).toBe('<div>helloworld</div>');
   expect(await page.$$eval(`text=lowo`, els => els.map(e => e.outerHTML).join(''))).toBe('<div>helloworld</div><span>helloworld</span>');
+
+  await page.setContent('<button id="testAttribute">Sign&nbsp;in</button>');
+  expect(await page.$eval('text="Sign in"', e => e.id)).toBe('testAttribute');
```
Comment (dgozman) [inline test/selectors-text.spec.ts:99]: Please also add a test for `$$` or `$$eval`.

File: test/selectors-text.spec.ts [hunk 3]
```diff
@@ -94,6 +94,9 @@ it('query', async ({page, isWebKit}) => {
   });
   expect(await page.$eval(`text=lowo`, e => e.outerHTML)).toBe('<div>helloworld</div>');
   expect(await page.$$eval(`text=lowo`, els => els.map(e => e.outerHTML).join(''))).toBe('<div>helloworld</div><span>helloworld</span>');
+
+  await page.setContent('<button id="testAttribute">Sign&nbsp;in</button>');
```
Comment (dgozman) [inline test/selectors-text.spec.ts:98]: Do we now match '\n' as well? Let's add a test. Also, do we want to collapse spaces, e.g. match "\n \n" by a single space? Maybe only in the relaxed mode without quotes?

**Comment by mxschmitt** [comment]
Superseded by #4312

**Pull request closed without merging**